apps/hpc-ftsadmin/src/app/components/tables/flows-table.tsx

```diff
@@ -0,0 +1,900 @@
+                        <Chip
+                          sx={chipSpacing}
+                          label={[
+                            t.t(lang, (s) => s.components.flowsTable.child),
+                          ]}
+                          size="small"
+                          color="primary"
+                        />
+                      )}
+                      {row.childIDs && row.childIDs.length > 0 && (
+                        <Chip
+                          sx={chipSpacing}
+                          label={[
+                            t.t(lang, (s) => s.components.flowsTable.parent),
+                          ]}
+                          size="small"
+                          color="primary"
+                        />
+                      )}
+                    </TableCell>
+                  );
+                case 'flow.newMoney':
+                  return (
+                    <TableCell
+                      key={`${row.id}v${row.versionID}_flow.newMoney`}
+                      size="small"
+                      data-test="flows-table-newMoney"
+                    >
+                      {(row.newMoney ?? '--').toString()}
+                    </TableCell>
+                  );
+                case 'flow.decisionDate':
+                  return (
+                    <TableCell
+                      key={`${row.id}v${row.versionID}_flow.decisionDate`}
+                      size="small"
+                      data-test="flows-table-decisionDate"
+                    >
+                      {row.decisionDate
+                        ? Intl.DateTimeFormat().format(
+                            new Date(row.decisionDate)
+                          )
+                        : '--'}
+                    </TableCell>
+                  );
+                case 'flow.exchangeRate':
+                  return (
+                    <TableCell
+                      key={`${row.id}v${row.versionID}_flow.exchangeRate`}
+                      size="small"
+                      data-test="flows-table-exchangeRate"
+                    >
+                      {row.exchangeRate ?? '--'}
+                    </TableCell>
+                  );
+                case 'flow.flowDate':
+                  return (
+                    <TableCell
+                      key={`${row.id}v${row.versionID}_flow.flowDate`}
+                      size="small"
+                      data-test="flows-table-flowDate"
+                    >
+                      {row.flowDate
+                        ? Intl.DateTimeFormat().format(new Date(row.flowDate))
+                        : '--'}
+                    </TableCell>
+                  );
+                case 'reportDetail.sourceID': {
+                  let rd = '--';
+                  if (row.reportDetails) {
+                    const uniqueSourceIDs = new Set(
+                      row.reportDetails.map((rd) => rd.sourceID)
+                    );
```

Comment:
This produces `Set<string | null | undefined>` and values `null` and `undefined` won't be skipped when you convert `Set` to array and call `join(', ')` below. If you run `Array.from(['115114', undefined, '-68117;-67234', null]).join(', ')`, you will see that result is `'115114, , -68117;-67234, '`.

Therefore, let's filter out values not defined.
```suggestion
                    const uniqueSourceIDs = new Set(
                      row.reportDetails.map((rd) => rd.sourceID).filter(util.isDefined)
                    );
```